glide-core/src/client/value_conversion.rs

```diff
@@ -6,8 +6,11 @@ use redis::{
 };
 
 #[derive(Clone, Copy)]
-pub(crate) enum ExpectedReturnType {
-    Map,
+pub(crate) enum ExpectedReturnType<'a> {
+    Map(
+        &'a Option<ExpectedReturnType<'a>>,
+        &'a Option<ExpectedReturnType<'a>>,
```

Comment:
```suggestion
        &'a Option<ExpectedReturnType<'a>>, // map key type
        &'a Option<ExpectedReturnType<'a>>, // map valye type
```
I assume you can omit the key type - it is always a string

Reply:
I'd prefer using curly brace syntax here to give names to the fields of the enum over just adding comments. Make the code a bit more self-documenting.
```suggestion
    Map {
        key_type: &'a Option<ExpectedReturnType<'a>>,
        value_type: &'a Option<ExpectedReturnType<'a>>,
    }
```